src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+    await waitForAnimationEnd(selectListElement);
+    await page.keyboard.press("ArrowDown");
+    await page.keyboard.press("ArrowDown");
+    await popoverContainerComponent(page).press("Escape");
+
+    await expect(selectListText(page, "green")).not.toBeVisible();
+    await expect(popoverContainerContent(page)).toBeVisible();
+  });
+
+  test("should close when the escape key is pressed with focus on the Select component", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PopoverContainerWithSelect />);
+
+    await popoverSettingsIcon(page).click();
+    const selectTextElement = selectText(page);
+    await selectTextElement.click();
+    await selectTextElement.press("Escape");
+    const popoverContainerContentElement = popoverContainerContent(page);
+
+    await expect(popoverContainerContentElement).not.toBeVisible();
+  });
+});
+
+test.describe("Event tests", () => {
+  test("should call onOpen callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <PopoverContainer
+        onOpen={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await expect(callbackCount).toBe(1);
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should call onOpen callback when a keyboard event is triggered by ${key} key`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <PopoverContainer
+          onOpen={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const popoverSettingsIconElement = popoverSettingsIcon(page);
+      await popoverSettingsIconElement.press(key);
+
+      await expect(callbackCount).toBe(1);
+    });
+  });
+
+  test("should call onClose callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <PopoverContainerComponent
+        onClose={() => {
+          callbackCount += 1;
+        }}
+        open
+      />
+    );
+
+    const popoverCloseIconElement = popoverCloseIcon(page);
+    await popoverCloseIconElement.click();
+
+    await expect(callbackCount).toBe(1);
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should call onClose callback when a ${key} keyboard event is triggered`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <PopoverContainerComponent
+          onClose={() => {
+            callbackCount += 1;
+          }}
+          open
+        />
+      );
+
+      const popoverCloseIconElement = popoverCloseIcon(page);
+      await popoverCloseIconElement.press(key);
+
+      await expect(callbackCount).toBe(1);
+    });
+  });
+
+  test("should call onClose callback when the escape is pressed", async ({
```

Comment:
```suggestion
  test("should call onClose callback when escape is pressed", async ({
```

**suggestion** (non-blocking): yes I know this was like this in the Cypress version too, but it reads oddly to me. Either remove "the", as I've suggested, or do "when the escape key is pressed" instead

Reply:
I prefer `when the escape key is pressed` so I'll go with that.